synfig-studio/src/gui/workarearenderer/renderer_guides.cpp

```diff
@@ -119,44 +105,129 @@ Renderer_Guides::render_vfunc(
-		{
-			const float y((*iter-window_starty)/ph);
-			if(iter==get_work_area()->curr_guide)
+			float x_temp = x_center;
+			float y_temp = y_center;
+
+			bool current_guide = false;
+			if(iter==get_work_area()->curr_guide){
 				cr->set_source_rgb(GDK_COLOR_TO_RGB(GUIDE_COLOR_CURRENT));
-			else
+				current_guide = true;
+			}
+			else{
 				cr->set_source_rgb(guides_color.get_r(),guides_color.get_g(),guides_color.get_b());
-
-			cr->move_to(
-				0,
-				y
+				current_guide = false;
+			}
+			if(iter->angle.get() != 0 && synfig::Angle::deg(iter->angle).get() != 90){
+
+				//draw the center of rotation for the selected guide
+				if (current_guide) {
+					cr->save();
+					cr->unset_dash();
+					cr->set_source_rgb(0,0,1.0);
+					cr->set_line_width(1.0);
+					cr->move_to(x_center + 6.0, y_center);
+					cr->line_to(x_center - 6.0, y_center);
+					cr->stroke();
+					cr->move_to(x_center, y_center + 6.0);
+					cr->line_to(x_center, y_center - 6.0);
+//					cr->set_source_rgb(0,1,0); //should it instead be a circle ?
+//					cr->arc(x_center, y_center, 3.0, 0, 6.82);
+					cr->stroke();
+					cr->restore();
+				}
+
+				//determine cordinate of point relative to center
+				std::string cordinate;
+				float slope = tan(iter->angle.get());
+				float angle = synfig::Angle::deg(iter->angle).get();
+				while (angle > 360)
+					angle -= 360;
+				if (angle > 0 && angle < 90)
+					cordinate="first";
+				else if (angle > 90 && angle < 180)
+					cordinate="second";
+				else if (angle > 180 && angle < 270)
+					cordinate = "third";
+				else
+					cordinate = "fourth";
+
+				if(cordinate == "first" || cordinate == "third" ){
+					while( (y_temp>0) && (x_temp<drawable_w)){
+						x_temp +=1;
+						y_temp += -1 *slope;
+
+				}
+				cr->move_to(x_center,y_center);
+				cr->line_to(
+					x_temp,
+					y_temp
+				);
+				cr->stroke();
+				cr->move_to(x_center,y_center);
+				while ((y_temp<drawable_h) && (x_temp>0)) {
+					x_temp -=1;
+					y_temp +=slope;
+					}
+				cr->line_to(
+					x_temp,
+					y_temp
 				);
-			cr->line_to(
-				drawable_w,
-				y
-			);
-			cr->stroke();
+				cr->stroke();
+			} else if(cordinate == "second" || cordinate == "fourth"){
+					while( (y_temp<drawable_h) && (x_temp<drawable_w)){
+						x_temp +=1;
+						y_temp -=slope;
+
+						}
+					cr->move_to(x_center,y_center);
+					cr->line_to(
+						x_temp,
+						y_temp
+					);
+					cr->stroke();
+					cr->move_to(x_center,y_center);
+
+					while( (y_temp>0) && (x_temp>0)){
+							x_temp -=1;
+							y_temp +=slope;
+					}
+					cr->line_to(
+						x_temp,
+						y_temp
+					);
+					cr->stroke();
+				}
+			} else {
+				if (iter->isVertical){
+					cr->move_to(
+						x_center,
+						0
+						);
+					cr->line_to(
+						x_center,
+						drawable_h
+					);
+					cr->stroke();
+				} else {
+					cr->move_to(
+						0,
+						y_center
+						);
+					cr->line_to(
+						drawable_w,
+						y_center
+					);
+					cr->stroke();
+				}
+			}
```

Comment:
You can do it simpler:
Give the guide line parameters (center + angle), you have the equation of the line.
Compute the line points at screen (vertical or horizontal) edges and draw the line that connects them.

Reply:
Yes, I thought of a doing something similar but haven't yet. This will better as we will find the two points which make the guide in one step instead of incrementally as it is right now. Will switch over to this implementation in a few hours.

Reply:
done.